qiskit/opflow/gradients/circuit_gradients/lin_comb.py

```diff
@@ -763,14 +827,56 @@ def _hessian_states(
-                                    # uses combo_fn on list op with a single operator
-                                    state = ListOp(
-                                        [state],
-                                        combo_fn=partial(self._hess_combo_fn, state_op=state_op),
-                                    )
+                                    if aux_meas_op == Z - 1j * Y:
+                                        state_z = ListOp(
+                                            [state],
+                                            combo_fn=partial(
+                                                self._hess_combo_fn, state_op=state_op
+                                            ),
+                                        )
+                                        pbc = PauliBasisChange(
+                                            replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                        )
+                                        pbc = pbc.convert(-Y ^ (I ^ (state.num_qubits - 1)))
+                                        state_y = pbc[-1] @ state
+                                        state_y = ListOp(
+                                            [state_y],
+                                            combo_fn=partial(
+                                                self._hess_combo_fn, state_op=state_op
+                                            ),
+                                        )
+                                        state = state_z - 1j * state_y
+
+                                    elif aux_meas_op == -Y:
+                                        pbc = PauliBasisChange(
+                                            replacement_fn=PauliBasisChange.measurement_replacement_fn
+                                        )
+                                        pbc = pbc.convert(
+                                            aux_meas_op ^ (I ^ (state.num_qubits - 1))
+                                        )
+                                        print(state)
+                                        state = pbc[-1] @ state
+                                        print(state)
+                                        state = -1 * ListOp(
+                                            [state],
+                                            combo_fn=partial(
+                                                self._hess_combo_fn, state_op=state_op
+                                            ),
+                                        )
+                                    elif aux_meas_op == Z:
+                                        state = ListOp(
+                                            [state],
+                                            combo_fn=partial(
+                                                self._hess_combo_fn, state_op=state_op
+                                            ),
+                                        )
```

Comment:
This code looks very similar to the fragment I commented on above. Can we extract and reuse the code to some extent?

Reply:
Ok